### ISSUE TITLE
`.` form can't handle repeated indexing

### ISSUE THREAD

**Issue by bb010g** [bakpakin/Fennel#45]
It would be nice if the `.` form could not only turn `(. foo bar)` into `foo[bar]`, but turn `(. foo bar baz)` into `foo[bar][baz]`. Similar functionality is provided by other forms, such as `..` turning `(.. foo bar baz)` into `foo .. bar .. baz` so you don't have to write long chains by hand.

An implementation of this as a macro isn't hard, but it seems useful enough that it should be built in:
```fnl
{".>" (fn [val ...]
        (var val val)
        (each [_ key (pairs [...])]
          (set val (list (sym ".") val key)))
        val)}
```

([Some discussion on Mastodon](https://a.weirder.earth/@bb010g/99749499483258097).)

**Comment by technomancy** [comment]
Yeah, I think it makes the most sense to make the primary `.` form
strict in that it requires each of the nested values (except the last)
to be a table. If you want to submit a PR for this I can review it.

A "short circuit" lookup that would simply return nil if it encountered
a nil at any point in the lookup would also be handy to have, but I
don't know if it should be included as a special form built-in to the
compiler as it could be done as a macro in a library.

**Comment by bakpakin** [comment]
Keep in mind the double eval problem if implementing the `.?` form. See #20 for a discussion of the problem.
